Automattic/jetpack
# Site Editor: Move the feature of changing backlink into mu-wpcom

## Proposed changes:


Moved the feature of changing the dahboard backlink from `plugins/jetpack` to `packages/jetpack-mu-wpcom` https://github.com/Automattic/jetpack/pull/33024/commits/fdfcc60c8634ea39005c82c9d11aa1f50c5b71e0 with some utility functions https://github.com/Automattic/jetpack/pull/33024/commits/e5eef45ec6fa450eca4020813561d8a31c248ca2 https://github.com/Automattic/jetpack/pull/33024/commits/f85d2b274e5ed9603eb7ed4b9f86e577cf19cff8. This is for keeping the wpcom specific feature into `jetpack-mu-wpcom`.

Fixes: https://github.com/Automattic/wp-calypso/issues/80595
Related: https://github.com/Automattic/jetpack/pull/32913

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



No

## Testing instructions:






* Apply the latest https://github.com/WordPress/gutenberg to your site (because this PR works with https://github.com/WordPress/gutenberg/pull/54242 and it's not released yet)
	* On Simple sites
		* Sync the change to your sandbox with AWE or rsync PCYsg-t5D-p2
	* On Atomic sites
		* Uninstall Gutenberg
		* Build the latest https://github.com/WordPress/gutenberg (`npm run dev`)
		* Create a new directory with including only `build/***`, `gutenberg.php`, `lib/***`, `packages/***`, `post-content.php`, `vendor/***`, and zip it
		* Upload the zip to `~/htdocs/wp-content/plugins` on your site 
		* Run `unzip gutenberg.zip` on your site and activate it
* Apply this patch to your site by following this comment: https://github.com/Automattic/jetpack/pull/33024#issuecomment-1717025149
	* Note that you must set `define( 'JETPACK_MU_WPCOM_LOAD_VIA_BETA_PLUGIN', true );` to your wp-config.php on Atomic sites.
* Go to the Theme Showcase 
* Click the Live Preview option on the three-dots menu on the Theme Card
* Confirm if the `<` link points to the Theme Showcase page

<img width="341" alt="Screen Shot 2023-09-07 at 15 05 41" src="https://github.com/Automattic/jetpack/assets/5287479/ceca0a51-5f54-41cf-b1c9-87a379e3dc6a">

In addition to the above, it'd be good to test;

- if nothing breaks in the "Normal" Site Editor 
- if nothing breaks in Launchpad
